src/converter/base.py

```diff
@@ -242,3 +246,74 @@ def masking(fig_node: dict) -> _Masking:
         "hasClippingMask": bool(fig_node.get("mask")),
         "clippingMaskMode": CLIPPING_MODE[fig_node.get("maskType", "OUTLINE")],
     }
+
+
+def parent_stack_mode(fig_node: dict) -> Optional[FlexDirection]:
+    # Get parent node if it exists
+    parent = None
+    if fig_node.get("parent"):
+        parent = context.fig_node(fig_node["parent"]["guid"])
+
+    # Return stack mode if parent has one
+    if parent and utils.has_auto_layout(parent):
+        if parent.get("stackMode") == "VERTICAL":
+            return FlexDirection.VERTICAL
+        else:
+            return FlexDirection.HORIZONTAL
+
+    return None
+
+
+def flex_item(fig_node: dict) -> Optional[FlexItem]:
+    if fig_node.get("stackPositioning") == "ABSOLUTE":
+        return FlexItem(ignoreLayout=True)
+
+    return None
+
+
+def horizontal_sizing_behaviour(fig_node: dict) -> SizingBehaviour:
+    stack_mode = parent_stack_mode(fig_node)
+
+    # Check condition for FIT behavior
+    is_resize_to_fit = fig_node.get("stackCounterSizing") == "RESIZE_TO_FIT_WITH_IMPLICIT_SIZE"
+    if is_resize_to_fit:
+        return SizingBehaviour.FIT
+
+    # Check conditions for FILL behavior
+    vertical_parent_with_stretch = (
+        stack_mode == FlexDirection.VERTICAL and fig_node.get("stackChildAlignSelf") == "STRETCH"
+    )
+
+    horizontal_parent_with_grow = stack_mode == FlexDirection.HORIZONTAL and fig_node.get(
+        "stackChildPrimaryGrow"
+    )
+
+    if vertical_parent_with_stretch or horizontal_parent_with_grow:
+        return SizingBehaviour.FILL
+
+    # Default behavior
+    return SizingBehaviour.FIXED
+
+
+def vertical_sizing_behaviour(fig_node: dict) -> SizingBehaviour:
+    stack_mode = parent_stack_mode(fig_node)
+
+    # Check conditions for FILL behavior
+    horizontal_parent_with_stretch = (
+        stack_mode == FlexDirection.HORIZONTAL and fig_node.get("stackChildAlignSelf") == "STRETCH"
+    )
+
+    vertical_parent_with_grow = stack_mode == FlexDirection.VERTICAL and fig_node.get(
+        "stackChildPrimaryGrow"
+    )
+
+    if horizontal_parent_with_stretch or vertical_parent_with_grow:
+        return SizingBehaviour.FILL
+
+    # Check condition for FIXED behavior
+    is_resize_to_fit = fig_node.get("stackPrimarySizing") == "FIXED"
+    if is_resize_to_fit:
+        return SizingBehaviour.FIXED
+
+    # Default behavior
+    return SizingBehaviour.FIT
```

Comment:
The sizing behaviours in Figma seem to have different defaults based on axis and configuration. This isn't the prettiest but it seems to be working reliably 🤔